packages/thirdweb/src/wallets/smart/index.ts

```diff
@@ -92,7 +93,8 @@ export async function connectSmartAccount(
   }
 
   const options = creationOptions;
-  const chain = connectChain ?? options.chain;
+  // Fallback to mainnet if no chain is provided (we only need this for pre-deploy signatures since transactions and deployments must define their own chain)
+  const chain = connectChain ?? options.chain ?? getCachedChain(1);
```

Comment:
One issue I see is that below this we get the factory and account contracts based on this chain variable.

It works with our default factory because it's deterministic (though need to double check it's on mainnet) but might error if you pass another factory address and no chain.

Also would double check we don't use this chain car for other assumptions in this connect flow